components/accept-financial-contributions/HostCollectiveCard.js

```diff
@@ -0,0 +1,273 @@
+import React, { useState, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Flex } from '@rebass/grid';
+import { FormattedDate, useIntl, defineMessages, FormattedMessage } from 'react-intl';
+import { graphql } from '@apollo/react-hoc';
+import { get } from 'lodash';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { getErrorFromGraphqlException } from '../../lib/errors';
+import { getCurrencySymbol } from '../../lib/utils';
+import { confettiFireworks } from '../../lib/confettis';
+import { Router } from '../../server/pages';
+
+import Container from '../Container';
+import { P, H1 } from '../Text';
+import StyledButton from '../StyledButton';
+import StyledCollectiveCard from '../StyledCollectiveCard';
+import Modal, { ModalBody, ModalHeader, ModalFooter } from '../StyledModal';
+import Avatar from '../Avatar';
+import HTMLContent from '../HTMLContent';
+import MessageBox from '../MessageBox';
+import StyledCheckbox from '../StyledCheckbox';
+import ExternalLink from '../ExternalLink';
+
+const messages = defineMessages({
+  collectives: {
+    id: 'pricingTable.row.collectives',
+    defaultMessage: 'Collectives',
+  },
+  contributors: {
+    id: 'ContributorsFilter.Financial',
+    defaultMessage: 'Financial contributors',
+  },
+  budget: {
+    id: 'YearlyBudget',
+    defaultMessage: 'Yearly budget',
+  },
+  apply: {
+    id: 'host.apply.create.btn',
+    defaultMessage: 'Apply',
+  },
+  currency: {
+    id: 'collective.currency.label',
+    defaultMessage: 'Currency',
+  },
+  hostFee: {
+    id: 'HostFee',
+    defaultMessage: 'Host fee',
+  },
+  hostSince: {
+    id: 'HostSince',
+    defaultMessage: 'Host since',
+  },
+  cancel: {
+    id: 'actions.cancel',
+    defaultMessage: 'Cancel',
+  },
+  submit: {
+    id: 'actions.submitApplication',
+    defaultMessage: 'Submit application',
+  },
+  tosError: {
+    id: 'acceptContributions.tosError',
+    defaultMessage: "Please accept the host's terms of service.",
+  },
+});
+
+const HostCollectiveCard = ({ host, collective, onChange, ...props }) => {
+  const [show, setShow] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [checked, setChecked] = useState(false);
+  const { formatMessage } = useIntl();
+
+  const date = new Date(host.createdAt);
+  const name = host.name;
+
+  const handleApplication = async () => {
+    if (get(host, 'settings.tos') && !checked) {
+      setError(formatMessage(messages.tosError));
+      return;
+    }
+    setLoading(true);
+    const collectiveInput = {
+      legacyId: collective.id,
+    };
+    const hostInput = {
+      legacyId: host.legacyId,
+    };
+    try {
+      await props.applyToHost(collectiveInput, hostInput);
+      Router.pushRoute('accept-financial-contributions', {
```

Comment:
I think it would be cleared if we kept it async.
```js
await Router.pushRoute(...);
window.scrollTo(0,0);
confettiFireworks(5000, { zIndex: 3000 });
```